src/Init/Prelude.lean

```diff
@@ -3361,22 +3361,20 @@ end Name
 
 /-- Source information of tokens. -/
 inductive SourceInfo where
-  | /--
-    Token from original input with whitespace and position information.
-    `leading` will be inferred after parsing by `Syntax.updateLeading`. During parsing,
-    it is not at all clear what the preceding token was, especially with backtracking.
-    -/
-   original (leading : Substring) (pos : String.Pos) (trailing : Substring) (endPos : String.Pos)
-  | /--
-    Synthesized syntax (e.g. from a quotation) annotated with a span from the original source.
-    In the delaborator, we "misuse" this constructor to store synthetic positions identifying
-    subterms.
-    -/
-    synthetic (pos : String.Pos) (endPos : String.Pos)
-  | /--
-    Synthesized token without position information.
-    -/
-    protected none
+  /--
+  Token from original input with whitespace and position information.
+  `leading` will be inferred after parsing by `Syntax.updateLeading`. During parsing,
+  it is not at all clear what the preceding token was, especially with backtracking.
+  -/
+  | original (leading : Substring) (pos : String.Pos) (trailing : Substring) (endPos : String.Pos)
+  /--
+  Synthesized syntax (e.g. from a quotation) annotated with a span from the original source.
+  In the delaborator, we "misuse" this constructor to store synthetic positions identifying
+  subterms.
+  -/
+  | synthetic (pos : String.Pos) (endPos : String.Pos)
+  /-- Synthesized token without position information. -/
+  protected | none
```

Comment:
This is maybe the only stylistic downside of moving all modifiers before the pipe.  We could split the doc comment and the other modifiers, but that requires modifying the `elabModifiers` code which is written in the old `stx[5][0]` style that I don't really want to touch unless necessary.

Moving the `protected` after the pipe again is a much smaller change that we can also do in a follow-up PR.

Reply:
AFAIR, this was the motivation for using the weird `| /-- ... -/ ctor` syntax: we avoided the weird `protected | none`

Reply:
What about the following hack?
- Leave `declModifiers` where it is.
- Add a `docComment` before `|`
- Return error at elaboration time if `declModifiers` contains a `docComment`.

Reply:
That sounds good to me, that's pretty much the same as the hack I suggested to handle `let rec` which has a similar issue

Reply:
> That sounds good to me, that's pretty much the same as the hack I suggested to handle `let rec` which has a similar issue

Yeah. Sorry, missed your comment at Zulip.

Reply:
Okay, I've implemented the "hacky" solution now.